src/Uno.UI.RemoteControl.Host/Helpers/CsprojUserGenerator.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using System.Globalization;
+using Microsoft.VisualStudio.SolutionPersistence;
+using Microsoft.VisualStudio.SolutionPersistence.Serializer;
+
+namespace Uno.UI.RemoteControl.Host.Helpers;
+
+/// <summary>
+/// Provides functionality to update or generate .csproj.user files for projects within a solution, setting the Uno
+/// Remote Control port for development tooling integration.
+/// </summary>
+public static class CsprojUserGenerator
+{
+	private static readonly XNamespace MsbuildNamespace = "http://schemas.microsoft.com/developer/msbuild/2003";
+
+	/// <summary>
+	/// Sets the user port configuration for all project files referenced by the specified solution file.
+	/// </summary>
+	/// <remarks>If the specified solution file does not exist or does not reference any projects, no changes are
+	/// made. Only solution files are supported; other file types are ignored.</remarks>
+	/// <param name="solution">The full path to an existing solution file (.sln or .slnx) whose referenced projects will have their user port
+	/// updated. Must not be null, empty, or point to a non-existent file.</param>
+	/// <param name="port">The port number to assign to each project's user configuration. Must be a valid TCP port number.</param>
+	public static async Task SetCsprojUserPort(string solution, int port)
+	{
+		// Only accept an existing solution file. Do not try to interpret other file types (e.g. .csproj).
+		if (string.IsNullOrWhiteSpace(solution) || !File.Exists(solution))
+		{
+			return; // Only use the provided path; do not search or treat it as a csproj.
+		}
+
+		var csprojUserPaths = new List<string>();
+		var extension = Path.GetExtension(solution);
+
+		if (
+			string.Equals(extension, ".sln", StringComparison.OrdinalIgnoreCase)
+			|| string.Equals(extension, ".slnx", StringComparison.OrdinalIgnoreCase))
+		{
+			csprojUserPaths.AddRange(await GetCsprojUserPathsFromSolutionAsync(solution));
```

Comment:
This means we will inject port on all project. Currently IDEs tries to do that only on "App" projects.